trlx/trainer/accelerate_base_trainer.py

```diff
@@ -425,10 +425,19 @@ def evaluate(self):  # noqa: C901
                 # in online setting, compute the reward for validation
                 if self.reward_fn:
                     logger.info("Computing rewards")
-                    rewards = torch.tensor(
-                        self.reward_fn(samples=str_samples, prompts=str_prompts, outputs=str_outputs, **metadata),
-                        dtype=float,
+                    rewards = self.reward_fn(
+                        samples=str_samples,
+                        prompts=str_prompts,
+                        outputs=str_outputs,
+                        model_tok=self.tokenizer,
+                        **metadata,
                     )
+                    if type(rewards[0]) is torch.Tensor:
+                        rewards = torch.tensor([reward.sum().item() for reward in rewards], dtype=float)
+                    elif type(rewards[0]) is list:
+                        rewards = torch.tensor([sum(reward) for reward in rewards])
+                    else:
+                        rewards = torch.tensor(rewards)
```

Comment:
Looking at the args into the `reward_fn`, changes highlighted:

```diff
  samples=str_samples,
  prompts=str_prompts,
  outputs=str_outputs,
+ model_tok=self.tokenizer,
  **metadata,
```

I'll assume that's intentional.

Looking at the args into the different calls to `torch.tensor`:

```diff
# Assume
raw_rewards = self.reward_fn(...)

# Convert raw to tensor rewards
rewards = torch.tensor(
                        raw_rewards,
-                        dtype=float, # Missing this?
                  )
```

Missing `dtype=float` on lines:
- ✅  436
- ❌ 438
- ❌ 440

Therefore, proposed fix:

```suggestion
                    raw_rewards = self.reward_fn(
                        samples=str_samples,
                        prompts=str_prompts,
                        outputs=str_outputs,
                        model_tok=self.tokenizer,
                        **metadata,
                    )
                    if type(rewards[0]) is torch.Tensor:
                        rewards = torch.tensor([reward.sum().item() for reward in rewards], dtype=float)
                    elif type(rewards[0]) is list:
                        rewards = torch.tensor([sum(reward) for reward in rewards], dtype=float)
                    else:
                        rewards = torch.tensor(rewards, dtype=float)
```


**Disclaimer**: I suck/new at Python (JavaScript dev), never used trlx (want to), so please ignore if these differences were intentional ❤️ .

Reply:
PR applying this fix: https://github.com/CarperAI/trlx/pull/520